R/metareactive.R

```diff
@@ -186,11 +231,15 @@ metaAction <- function(expr, env = parent.frame(), quoted = FALSE) {
   # Need to wrap expr with shinymeta:::metaExpr, but can't use rlang/!! to do
   # so, because we want to keep any `!!` contained in expr intact (i.e. too
   # early to perform expansion of expr here).
-  expr <- wrapExpr(shinymeta::metaExpr, expr, env)
+  expr <- wrapExpr(shinymeta::metaExpr, expr)
 
-  function() {
-    rlang::eval_tidy(expr, NULL, env)
+  func <- function() {
+    eval(expr, envir = env)
```

Comment:
Maybe we should if you try to evaluate this expr more than once outside of meta mode?

Reply:
You know, that occurred to me too, but then I thought there might be legitimate reasons to use this more than once. (It's harder for me to imagine why you might have `run=TRUE` and also run more than once, so we could single out that case with a warning, but it seemed a bit too arbitrary to me.)

Reply:
I agree that it's arbitrary, but it's going to be the most common mistake

Reply:
Another way to go is to remove `run=TRUE`, have it always run immediately, and make any subsequent (non-metamode) run always an error? I believe you get similar behavior if you try to, say, run a `metaObserver` outside of meta mode.

We can wait for legitimate use cases for `run=FALSE` to appear, and then either add a new argument or introduce a new function.

Reply:
Sounds good to me!